pytest_catchlog.py

```diff
@@ -162,59 +190,96 @@ def record_tuples(self):
 
             (logger_name, log_level, message)
         """
-        return [(r.name, r.levelno, r.getMessage()) for r in self.records()]
-
-    def set_level(self, level, logger=None):
-        """Sets the level for capturing of logs.
-
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
-        """
+        return [(r.name, r.levelno, r.getMessage()) for r in self.buffer]
 
-        obj = logger and logging.getLogger(logger) or self.handler
-        obj.setLevel(level)
+    @property
+    def buffer(self):
+        return self._handler.buffer
 
-    def at_level(self, level, logger=None):
-        """Context manager that sets the level for capturing of logs.
+    @property
+    def lines(self):
+        return [r.getMessage() for r in self.buffer]
 
-        By default, the level is set on the handler used to capture
-        logs. Specify a logger name to instead set the level of any
-        logger.
+    def __init__(self, handler):
+        # Do not call LineMatcher.__init__(),
+        # it attempts to assign to self.lines, which is a property.
```

Comment:
Good point - I didn't notice it was actually documented. I still think it's probably intended for internal pytest use and the documentation is intended for hooks returning a `LineMatcher` - but hey, what can possibly go wrong.

On a second thought, I'd say let's keep this and see if it breaks - which, after all, is unlikely.